subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/JavaPluginIntegrationTest.groovy

```diff
@@ -61,6 +61,19 @@ class JavaPluginIntegrationTest extends AbstractIntegrationSpec {
         then:
         file("build/classes/java/main").assertDoesNotExist()
         file("build/classes/main/Main.class").assertExists()
-        result.assertOutputContains("Using a single directory for all classes from a source set.")
+    }
+
+    def "emits deprecation message if something uses classesDir"() {
```

Comment:
Not sure if we need to test that but the `test` task would also have the status `NO-SOURCE`.